control-plane/rest/openapi-specs/v0_api_spec.yaml

```diff
@@ -2334,6 +2371,7 @@ components:
         thin: false
         topology: null
         affinity_group: null
+        max_snapshots: -1
```

Comment:
Changed to logic as follow.
1. If max_snapshot field is not passed in storage class, then it is considered as unlimited else it will consider the storage class parameter value to judge max snapshot limit.
2. Using plugin, We can change the value of max_snapshot field value.
3. Parameter added in core-agent argument is removed now.